src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -113,6 +113,26 @@ public void SetAttributes(string path, FileAttributes attributes)
             if (!_exists)
                 FileSystemInfo.ThrowNotFound(path);
 
+            if (Interop.Sys.CanSetHiddenFlag)
+            {
+                if ((attributes & FileAttributes.Hidden) != 0)
+                {
+                    if ((_fileStatus.UserFlags & (uint)Interop.Sys.UserFlags.UF_HIDDEN) == 0)
+                    {
+                        // If Hidden flag is set and cached file status does not have the flag set then set it
+                        Interop.CheckIo(Interop.Sys.LChflags(path, (uint)Interop.Sys.UserFlags.UF_HIDDEN), path, InitiallyDirectory);
```

Comment:
For robustness, shouldn't this be passing `_fileStatus.UserFlags | (uint)Interop.Sys.UserFlags.UF_HIDDEN` instead of just `(uint)Interop.Sys.UserFlags.UF_HIDDEN`?